### ISSUE TITLE
bug: Symfony.Formatting.ReturnOrThrow detects false positive for closures

### ISSUE THREAD

**Issue by xalopp** [djoos/Symfony-coding-standard#118]
Unfortunately, the following false positive 
```
 20 | ERROR   | Do not use else, elseif, break after if and case conditions which return or throw something (Symfony.Formatting.ReturnOrThrow.Invalid)

```
pops up for closure, as you can see in this code

```php
<?php

namespace Fail;

/**
 * Class PortalDataService
 */
class Clazz
{
    /**
     * @param array $arr
     *
     * @return array
     */
    public function clear($arr)
    {
        if (null !== $arr) {
            $mapped = $arr->map(
                function ($e) {
                    return '';
                }
            )->toArray();
        } else {
            $mapped = [];
        }

        return $mapped;
    }
}
```

This is a false positive, since the `return` belongs to the closure an not the function.

**Comment by gseric** [comment]
Same problem with return somewhere inside try/catch:

```php
if (true) {
    try {
        // ...
    } catch (Exception $e) {
        return false;
    }
} else {
    // ...
}
```

"else" is falsely considered as unnecessary...

**Comment by djoos** [comment]
#126 merged!
